src/OrchardCore.Modules/OrchardCore.Notifications/Activities/NotifyUserTask.cs

```diff
@@ -1,41 +1,61 @@
+using System;
 using System.Collections.Generic;
-using System.Linq;
 using System.Text.Encodings.Web;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Localization;
 using Microsoft.Extensions.Logging;
 using OrchardCore.Users;
+using OrchardCore.Users.Indexes;
 using OrchardCore.Users.Models;
 using OrchardCore.Workflows.Models;
 using OrchardCore.Workflows.Services;
+using YesSql;
+using YesSql.Services;
 
 namespace OrchardCore.Notifications.Activities;
 
 public class NotifyUserTask : NotifyUserTaskActivity<NotifyUserTask>
 {
+    private readonly ISession _session;
+
     public NotifyUserTask(
        INotificationService notificationCoordinator,
        IWorkflowExpressionEvaluator expressionEvaluator,
        HtmlEncoder htmlEncoder,
        ILogger<NotifyUserTask> logger,
-       IStringLocalizer<NotifyUserTask> localizer
+       IStringLocalizer<NotifyUserTask> stringLocalizer,
+       ISession session
    ) : base(notificationCoordinator,
        expressionEvaluator,
        htmlEncoder,
        logger,
-       localizer)
+       stringLocalizer)
     {
+        _session = session;
     }
 
     public override LocalizedString DisplayText => S["Notify User Task"];
 
-    protected override Task<IEnumerable<IUser>> GetUsersAsync(WorkflowExecutionContext workflowContext, ActivityContext activityContext)
+    public WorkflowExpression<string> Usernames
```

Comment:
@MikeAlhayek 
I rather keep it as `username` UserName to me is the name of the user like "Mike Alhayek". Username is an identifier (for example, malhayek) so it would be Username.

Reply:
The casing is `UserName` for this identifier otherwise, including in `UserIndex` where it references it.

Reply:
If you search our code you'll see that Username is used in most cases not UserName. From a language perspective, I think the correct test for a unique identifier is `username` where `UserName` (User Name) referees to the user's name

Reply:
The only place where the name of the user account is not called UserName in members is in the `UsernameOrEmail` members of the forgot and reset password features:

![image](https://github.com/OrchardCMS/OrchardCore/assets/1976647/782b09f0-d0d2-4fc6-9205-1f0f6c0020a6)

"UserName" turns up 322 matches while "userName" 119. It's "UserName" as the property of `User` too, what all of the changes in this PR refer to.

This includes `UserIndex` what this PR uses, so there's "_session.Query<User, UserIndex>(user => user.NormalizedUser**N**ame.IsIn(user**n**ames))."

We only have this kind of name, i.e. the unique identifier, in the system. While we could have a linguistic debate if this were the start of the project, and I don't have a strong feeling either way, it is definitely "UserName" currently.

Let's just keep the consistency.

Reply:
I guess if we are wrong, let's just be wrong "almost" everywhere. I made this change